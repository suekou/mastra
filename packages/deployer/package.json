{
  "name": "@mastra/deployer",
  "version": "0.2.4-alpha.0",
  "description": "",
  "type": "module",
  "files": [
    "dist"
  ],
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "exports": {
    ".": {
      "import": {
        "types": "./dist/index.d.ts",
        "default": "./dist/index.js"
      },
      "require": {
        "types": "./dist/index.d.cts",
        "default": "./dist/index.cjs"
      }
    },
    "./server": {
      "import": {
        "types": "./dist/server/index.d.ts",
        "default": "./dist/server/index.js"
      },
      "require": {
        "types": "./dist/server/index.d.cts",
        "default": "./dist/server/index.cjs"
      }
    },
    "./build": {
      "import": {
        "types": "./dist/build/index.d.ts",
        "default": "./dist/build/index.js"
      },
      "require": {
        "types": "./dist/build/index.d.cts",
        "default": "./dist/build/index.cjs"
      }
    },
    "./bundler": {
      "import": {
        "types": "./dist/bundler/index.d.ts",
        "default": "./dist/bundler/index.js"
      },
      "require": {
        "types": "./dist/bundler/index.d.cts",
        "default": "./dist/bundler/index.cjs"
      }
    },
    "./analyze": {
      "import": {
        "types": "./dist/build/analyze.d.ts",
        "default": "./dist/build/analyze.js"
      },
      "require": {
        "types": "./dist/build/analyze.d.cts",
        "default": "./dist/build/analyze.cjs"
      }
    },
    "./loader": {
      "import": {
        "types": "./dist/validator/loader.d.ts",
        "default": "./dist/validator/loader.js"
      },
      "require": {
        "types": "./dist/validator/loader.d.cts",
        "default": "./dist/validator/loader.cjs"
      }
    },
    "./package.json": "./package.json"
  },
  "scripts": {
    "build": "tsup src/index.ts src/build/index.ts src/server/index.ts src/build/bundler.ts src/build/analyze.ts src/bundler/index.ts src/validator/loader.ts  src/validator/custom-resolver.ts --format esm,cjs --clean --experimental-dts --treeshake=smallest --splitting --publicDir",
    "build:watch": "pnpm build --watch",
    "pull:openapispec": "node src/server/openapi.script.js",
    "test": "vitest run",
    "lint": "eslint ."
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@babel/core": "^7.26.9",
    "@babel/helper-module-imports": "^7.25.9",
    "@mastra/core": "workspace:^",
    "@neon-rs/load": "^0.1.82",
    "@rollup/plugin-alias": "^5.1.1",
    "@rollup/plugin-commonjs": "^28.0.3",
    "@rollup/plugin-json": "^6.1.0",
    "@rollup/plugin-node-resolve": "^16.0.0",
    "@rollup/plugin-virtual": "^3.0.2",
    "builtins": "^5.1.0",
    "detect-libc": "^2.0.3",
    "dotenv": "^16.4.7",
    "esbuild": "^0.25.1",
    "fs-extra": "^11.3.0",
    "hono": "^4.7.4",
    "resolve-from": "^5.0.0",
    "rollup": "^4.35.0",
    "rollup-plugin-esbuild": "^6.2.1",
    "rollup-plugin-node-externals": "^8.0.0",
    "typescript-paths": "^1.5.1",
    "zod": "^3.24.2"
  },
  "devDependencies": {
    "@hono/node-server": "^1.13.8",
    "@hono/swagger-ui": "^0.5.1",
    "@internal/lint": "workspace:*",
    "@microsoft/api-extractor": "^7.52.1",
    "@types/babel__core": "^7.20.5",
    "@types/babel__helper-module-imports": "^7.18.3",
    "@types/fs-extra": "^11.0.4",
    "@types/node": "^22.13.10",
    "concurrently": "^9.1.2",
    "eslint": "^9.22.0",
    "hono-openapi": "^0.4.6",
    "rollup-plugin-visualizer": "^5.14.0",
    "superjson": "^2.2.2",
    "tsup": "^8.4.0",
    "type-fest": "^4.37.0",
    "typescript": "^5.8.2",
    "vitest": "^2.1.9",
    "zod-to-json-schema": "^3.24.3"
  }
}
