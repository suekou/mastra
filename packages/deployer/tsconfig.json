{
  "extends": "../../tsconfig.node.json",
  "include": ["./global.d.ts", "src/**/*"],
  "exclude": ["node_modules", "**/*.test.ts", "**/__fixtures__/"]
}
