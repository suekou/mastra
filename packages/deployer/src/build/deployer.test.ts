import { describe, it, expect } from 'vitest';
import { fileURLToPath } from 'node:url';
import { dirname, join } from 'node:path';
import { getDeployerBundler } from './deployer';

describe('getDeployer', () => {
  const _dirname = dirname(fileURLToPath(import.meta.url));

  it.for([
    ['./plugins/__fixtures__/basic.js'],
    ['./plugins/__fixtures__/basic-with-const.js'],
    ['./plugins/__fixtures__/basic-with-import.js'],
    ['./plugins/__fixtures__/basic-with-function.js'],
    ['./plugins/__fixtures__/mastra-with-extra-code.js'],
    ['./plugins/__fixtures__/empty-mastra.js'],
  ])('should be able to extract the deployer from %s', async ([fileName]) => {
    const bundle = await getDeployerBundler(join(_dirname, fileName));

    const result = await bundle.generate({
      format: 'esm',
    });

    expect(result?.output[0].code).toMatchSnapshot();
  });
});
