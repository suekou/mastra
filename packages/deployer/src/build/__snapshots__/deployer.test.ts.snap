// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getDeployer > should be able to extract the deployer from ./plugins/__fixtures__/basic.js 1`] = `
"import { TestDeployer } from '@mastra/deployer/test';

const deployer = new TestDeployer();

export { deployer };
"
`;

exports[`getDeployer > should be able to extract the deployer from ./plugins/__fixtures__/basic-with-const.js 1`] = `
"import { TestDeployer } from '@mastra/deployer/test';

const testDeployer = new TestDeployer();
const deployer = testDeployer;

export { deployer };
"
`;

exports[`getDeployer > should be able to extract the deployer from ./plugins/__fixtures__/basic-with-function.js 1`] = `
"import { testDeployer } from '@mastra/deployer/test';

function getDeployer() {
  return testDeployer;
}
const deployer = getDeployer();

export { deployer };
"
`;

exports[`getDeployer > should be able to extract the deployer from ./plugins/__fixtures__/basic-with-import.js 1`] = `
"import { testDeployer } from '@mastra/deployer/test';

const deployer = testDeployer;

export { deployer };
"
`;

exports[`getDeployer > should be able to extract the deployer from ./plugins/__fixtures__/empty-mastra.js 1`] = `
"
"
`;

exports[`getDeployer > should be able to extract the deployer from ./plugins/__fixtures__/mastra-with-extra-code.js 1`] = `
"import { testDeployer } from '@mastra/deployer/test';

function getDeployer() {
  return testDeployer;
}
const deployer = getDeployer();

export { deployer };
"
`;
